src/main/frontend/src/components/landingPage/InfiniteScrollComponent.js

```diff
@@ -0,0 +1,66 @@
+import InfiniteScroll from "react-infinite-scroll-component";
+import { useEffect, useState } from "react";
+import Item from "./Item";
+import { Row, Col } from "react-bootstrap";
+import { fetchLastChance, fetchNewArrivals } from './ItemService';
+
+function InfiniteScrollComponent(props) {
+  const [items, setItems] = useState([]);
+  const [hasMoreItems, setHasMoreItems] = useState(false);
+  const [page, setPage] = useState(0);
+
+  useEffect(async () => {
+    let data = "";
+
+    if (props.load === "newarrivals") {
+      data = await fetchNewArrivals(page, 4);
+    } else {
+      data = await fetchLastChance(page, 4);
+    }
+
+    setItems(data.content);
+  }, []);
+
+  const fetchData = async () => {
+    let itemsFromServer = "";
+
+    if (props.load === "newarrivals") {
+      itemsFromServer = await fetchNewArrivals(page+1, 4);
+    } else {
+      itemsFromServer = await fetchLastChance(page+1, 4);
+    }
+
+    setItems([...items, ...itemsFromServer.content]);
+    setHasMoreItems(itemsFromServer.last);
```

Comment:
This can be moved to new useEffect that will have page in dependency array, and fetch data will only need to setPage(page + 1); That way, first useEffect can be removed.